File: Explore/2020April/Week4/C/jumpGame.c
bool canJump(int* nums, int numsSize){
    int target = 0;
    
    for(int i = 0; i < numsSize; i++)
    {
        if(i > target)
            return false;
        
        if(target < i + nums[i])
            target = i + nums[i];
    }
    
    return true;
}

File: Explore/2020April/Week4/Python/jumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        target = 0
        
        for i, num in enumerate(nums):
            if i > target:
                return False
            print(i, num, target)
            target = max(target, i + num)
            
        return True
